src/EnergyPlus/SizingManager.cc

```diff
@@ -2177,16 +2246,31 @@ void DetermineSystemPopulationDiversity(EnergyPlusData &state)
 
     // compute D for standard 62.1 by system
     for (int AirLoopNum = 1; AirLoopNum <= state.dataHVACGlobal->NumPrimaryAirSys; ++AirLoopNum) {
-        if (state.dataSize->PzSumBySys(AirLoopNum) > 0.0) {
-            state.dataSize->DBySys(AirLoopNum) = state.dataSize->PsBySys(AirLoopNum) / state.dataSize->PzSumBySys(AirLoopNum);
+        int SysSizNum = UtilityRoutines::FindItemInList(
+            FinalSysSizing(AirLoopNum).AirPriLoopName, state.dataSize->SysSizInput, &SystemSizingInputData::AirPriLoopName);
+        if (SysSizNum == 0) SysSizNum = 1; // use first when none applicable
+
+        // compute D if set to autosize
+        if (SysSizInput(SysSizNum).OccupantDiversity == AutoSize) {
+            if (state.dataSize->PzSumBySys(AirLoopNum) > 0.0) {
+                state.dataSize->DBySys(AirLoopNum) = state.dataSize->PsBySys(AirLoopNum) / state.dataSize->PzSumBySys(AirLoopNum);
+            } else {
+                state.dataSize->DBySys(AirLoopNum) = 1.0;
+            }
+            state.dataSize->DBySys(AirLoopNum) = min(1.0, state.dataSize->DBySys(AirLoopNum));
         } else {
-            state.dataSize->DBySys(AirLoopNum) = 1.0;
+            // set the occupant diversity based on user-specified value
+            state.dataSize->DBySys(AirLoopNum) = SysSizInput(SysSizNum).OccupantDiversity;
         }
-        state.dataSize->DBySys(AirLoopNum) = min(1.0, state.dataSize->DBySys(AirLoopNum));
 
         // For single zone systems, D should be 1.0.
         if (state.dataAirLoop->AirLoopZoneInfo(AirLoopNum).NumZones == 1) {
             state.dataSize->DBySys(AirLoopNum) = 1.0;
+            ShowWarningError(
+                state,
+                format("The {} air loop serves a single zone. The Occupant Diversity was calculated or set to a value less than 1.0. Single-zone air "
+                       "loops should have an Occupant Diversity of 1.0. The Occupant Diversity value for that air loop has been reset to 1.0",
+                       FinalSysSizing(AirLoopNum).AirPriLoopName));
```

Comment:
Add new warning when the Occupant Diversity is set to a value lower than 1.0 but the system is a single zone system.